### PR TITLE
Feature/macos push

### PR DESCRIPTION
**Pull request by jaelim-ms**

### CODE REVIEW

File: MobileCenterCrashes/MobileCenterCrashesTests/MSErrorLogFormatterTests.mm [hunk 1]
```diff
@@ -46,6 +46,13 @@ - (void)testCreateErrorReport {
   assertThat(errorReport.appErrorTime, equalTo(crashReport.systemInfo.timestamp));
   assertThat(errorReport.appStartTime, equalTo(crashReport.processInfo.processStartTime));
 
+  /*
+   * FIXME: Crashes will look up a session from history to get an appropriate session at crash time. This causes
+   * intermittent failures due to osVersion mismatch depends on running sequence of the tests. If there is a history
+   * that was built by other tests, device property might have mocked data and this test will be failed. To prevent this
```
Comment (TroubleMakerBen) [inline MobileCenterCrashes/MobileCenterCrashesTests/MSErrorLogFormatterTests.mm]: `this test will fail` sounds better.

File: MobileCenterCrashes/MobileCenterCrashesTests/MSErrorLogFormatterTests.mm [hunk 2]
```diff
@@ -64,6 +71,13 @@ - (void)testCreateErrorReport {
...
```
Comment (TroubleMakerBen) [inline MobileCenterCrashes/MobileCenterCrashesTests/MSErrorLogFormatterTests.mm]: Same as above: `will fail` sounds better to me.

File: MobileCenterPush/MobileCenterPush/MSPush.m [hunk 3]
```diff
@@ -192,29 +215,80 @@ - (void)didFailToRegisterForRemoteNotificationsWithError:(NSError *)error {
...
+  return NO;
+}
+#endif
+
+- (BOOL)didReceiveRemoteNotification:(NSDictionary *)userInfo fromUserNotification:(BOOL)userNotification {
+
+#if !TARGET_OS_OSX
+  (void)userNotification;
+#endif
   MSLogVerbose([MSPush logTag], @"User info for notification has forwarded to Push: %@", [userInfo description]);
```
Comment (TroubleMakerBen) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: This was not changed in this PR, but I think it should say `User info for notification was forwarded...`.

Comment (jaelim-ms) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: I'll submit a separate PR from develop branch base to avoid merge pains.

Comment (jaelim-ms) [inline MobileCenterPush/MobileCenterPush/MSPush.m]: There is a PR submitted for this change. Look at #699.

**Pull request merged**